src/runtime/loader.ts

```diff
@@ -63,16 +63,24 @@ export class Loader {
     if (/^https?:\/\//.test(file)) {
       return this._loadURL(file);
     }
-    return this._loadFile(file);
+    return this._loadFile(file, 'utf-8') as Promise<string>;
   }
 
-  async _loadFile(file: string): Promise<string> {
+  async loadBinary(file: string): Promise<ArrayBuffer> {
+    if (/^https?:\/\//.test(file)) {
+      return fetch(file).then(res => res.arrayBuffer());
+    } else {
+      return this._loadFile(file, null) as Promise<ArrayBuffer>;
+    }
+  }
+
+  async _loadFile(file: string, encoding?: string): Promise<string | ArrayBuffer> {
```

Comment:
I recently learned you can do something like:

```
interface Mapping {
  'utf-8': string;
  'binary': ArrayBuffer;
}
async _loadFile<T extends keyof Mapping>(file: string, encoding: T): Promise<Mapping[T]> {
  ...
}
```

Reply:
Does that work for a null input? `encoding` is actually either 'utf-8' or null.

... but even if it does, I'm not sure that's better. The API itself is a lot less readable IMO.

Reply:
It does work with null. I would do it for type safety and getting rid of casting and all that.

I agree it's a bit verbose, but I would still take it for type safety.

Use your best judgement.